### PR TITLE
On line vectorpostprocessor migration

### LINKED ISSUE #7091: Migration of codes for on line quantity calculation
### Description of the enhancement or error report

Migrated AccumulateAux and LineMaterialSymmTensorSampler (LineMaterialRankTwoSampler and LineMaterialRankTwoScalarSampler), deprecated ElementsOnLineAux, MaterialTensorOnLine and removed the tests, and updated related input file syntax, changed the parameter type for curr_point from pointer to reference in both MaterialTensorCalculator and LineMaterialSamplerBase and their usage in other classes

### PR DESCRIPTION
**Pull request by hchen139**
closes #7091

### CODE REVIEW

File: modules/tensor_mechanics/include/userobjects/RankTwoOnLine.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+#include "ElementUserObject.h"
+#include "RankTwoAux.h"
+
+class RankTwoOnLine;
+class RankTwoTensor;
+
+template<>
+InputParameters validParams<RankTwoOnLine>();
+
```
Comment (sapitts) [inline modules/tensor_mechanics/include/userobjects/RankTwoOnLine.h]: Add in the Doxygen comments in here to explain what this class does--touch base with Ben if need be

File: modules/tensor_mechanics/include/auxkernels/ElementsOnLineAux.h [hunk 2]
```diff
@@ -9,31 +9,29 @@
...
+InputParameters validParams<ElementsOnLineAux>();
+
 class ElementsOnLineAux : public AuxKernel
 {
-
 public:
-
   ElementsOnLineAux(const InputParameters & parameters);
 
   virtual ~ElementsOnLineAux() {}
```
Comment (dschwen) [inline modules/tensor_mechanics/include/auxkernels/ElementsOnLineAux.h]: not necessary

File: modules/tensor_mechanics/src/userobjects/RankTwoOnLine.C [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+RankTwoOnLine::initialize()
+{
+  _dist.clear();
+  _value.clear();
+}
+
+void
+RankTwoOnLine::execute()
+{
+  unsigned int qp(0); // all integration points have the same _elem_line_id, just use qp = 0
```
Comment (dschwen) [inline modules/tensor_mechanics/src/userobjects/RankTwoOnLine.C]: use `=` here

File: modules/tensor_mechanics/src/userobjects/RankTwoOnLine.C [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+  params.addParam<MooseEnum>("quantity_type", quantity_options, "Type of output quantity");
+  params.addRangeCheckedParam<unsigned int>("index_i", 0, "index_i >= 0 & index_i <= 2", "The index i of ij for the tensor to output (0, 1, 2)");
+  params.addRangeCheckedParam<unsigned int>("index_j", 0, "index_j >= 0 & index_j <= 2", "The index j of ij for the tensor to output (0, 1, 2)");
+  params.addParam<Point>("point1", Point(0, 0, 0), "Start point of line along which material data is output");
+  params.addParam<Point>("point2", Point(0, 1, 0), "End point of line along which material data is output");
+  params.addParam<Point>("direction", Point(0, 0, 1), "Direction vector");
+  params.addParam<int>("line_id", 1, "ID of the line of elements to output stresses on");
+  params.addRequiredParam<std::string>("filename", "Output file name");
+  params.addCoupledVar("element_line_id", "Element line ID: if not zero, output stress at integration points");
```
Comment (dschwen) [inline modules/tensor_mechanics/src/userobjects/RankTwoOnLine.C]: Doc is unclear

File: modules/tensor_mechanics/src/userobjects/RankTwoOnLine.C [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+#include "RankTwoOnLine.h"
+#include "RankTwoScalarTools.h"
+
+// libmesh includes
+#include "libmesh/quadrature.h"
+
+template<>
+InputParameters validParams<RankTwoOnLine>()
+{
+  InputParameters params = validParams<ElementUserObject>();
```
Comment (dschwen) [inline modules/tensor_mechanics/src/userobjects/RankTwoOnLine.C]: This whole thing should probably be rewritten as a `VectorPostprocessor`!

File: modules/tensor_mechanics/src/userobjects/RankTwoOnLine.C [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+  MooseEnum quantity_options("Component VonMisesStress EquivalentPlasticStrain Hydrostatic L2norm MaxPrincipal MidPrincipal MinPrincipal VolumetricStrain FirstInvariant SecondInvariant ThirdInvariant AxialStress HoopStress RadialStress TriaxialityStress Direction");
+  params.addParam<MooseEnum>("quantity_type", quantity_options, "Type of output quantity");
+  params.addRangeCheckedParam<unsigned int>("index_i", 0, "index_i >= 0 & index_i <= 2", "The index i of ij for the tensor to output (0, 1, 2)");
+  params.addRangeCheckedParam<unsigned int>("index_j", 0, "index_j >= 0 & index_j <= 2", "The index j of ij for the tensor to output (0, 1, 2)");
+  params.addParam<Point>("point1", Point(0, 0, 0), "Start point of line along which material data is output");
+  params.addParam<Point>("point2", Point(0, 1, 0), "End point of line along which material data is output");
+  params.addParam<Point>("direction", Point(0, 0, 1), "Direction vector");
+  params.addParam<int>("line_id", 1, "ID of the line of elements to output stresses on");
+  params.addRequiredParam<std::string>("filename", "Output file name");
```
Comment (dschwen) [inline modules/tensor_mechanics/src/userobjects/RankTwoOnLine.C]: use `FileName`

File: modules/tensor_mechanics/src/userobjects/RankTwoOnLine.C [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+
+  params.addRequiredParam<MaterialPropertyName>("rank_two_tensor", "The rank two material tensor name");
+  MooseEnum quantity_options("Component VonMisesStress EquivalentPlasticStrain Hydrostatic L2norm MaxPrincipal MidPrincipal MinPrincipal VolumetricStrain FirstInvariant SecondInvariant ThirdInvariant AxialStress HoopStress RadialStress TriaxialityStress Direction");
+  params.addParam<MooseEnum>("quantity_type", quantity_options, "Type of output quantity");
+  params.addRangeCheckedParam<unsigned int>("index_i", 0, "index_i >= 0 & index_i <= 2", "The index i of ij for the tensor to output (0, 1, 2)");
+  params.addRangeCheckedParam<unsigned int>("index_j", 0, "index_j >= 0 & index_j <= 2", "The index j of ij for the tensor to output (0, 1, 2)");
+  params.addParam<Point>("point1", Point(0, 0, 0), "Start point of line along which material data is output");
+  params.addParam<Point>("point2", Point(0, 1, 0), "End point of line along which material data is output");
+  params.addParam<Point>("direction", Point(0, 0, 1), "Direction vector");
+  params.addParam<int>("line_id", 1, "ID of the line of elements to output stresses on");
```
Comment (dschwen) [inline modules/tensor_mechanics/src/userobjects/RankTwoOnLine.C]: Doc is completely unclear. What is a "line ID"?

**Comment by hchen139** [comment]
@dschwen Do you have any idea why cstdlib can not be found?

**Comment by dschwen** [comment]
Nope. I've invalidated one test. Let's see.

**Comment by dschwen** [comment]
So... what's going on here?

**Pull request merged**